authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -355,12 +375,21 @@ injectOAuthToCred oa cred =
             ] cred
 
 
+-- Note that prefix is used for realm in addAuthHeader, so should fix that up,
+-- but may need an API break for that.
+addAuthBody :: BS.ByteString -> Credential -> Request -> Request
+addAuthBody _ (Credential cred) req = urlEncodedBody (filterCreds cred) req
```

Comment:
I'd be happy to. I don't know how to do that and still support both of these as valid calls:

a) convert a temporary token to an access token on a site using section 5.2.2 POST body encoding (e.g. Launchpad only accepts this form):
mytoken <- getAccessToken'' addAuthBody id some_oauth temp_credential manager

b) convert a temporary token to an access token on a site using section 5.2.1 Authorisation header (e.g. Github)
mytoken <- getAccessToken'' addAuthHeader id some_oauth temp_credential manager

If I remove the unused parameter from addAuthBody then the first will get a type error. If I change the type for the parameter to mytoken then I'll get a type error in b) using it with addAuthHeader. the parameter that you're asking about is used with addAuthHeader, its just a not used for addAuthBody.

If I change the signature of addAuthHeader, or add an addAuthHeader' then I can make it the same as addAuthBoth needs, by pushing the realm logic down into the auth attachment helper (which is where I think it belongs btw, just like I said, didn't want to break the API for addAuthHeader).